enhancements/improve-kubeletconfig-default-configuration-check-enhancement.md

```diff
@@ -0,0 +1,278 @@
+  "podPidsLimit": 4096,
+  "resolvConf": "/etc/resolv.conf",
+  "cpuCFSQuota": true,
+  "cpuCFSQuotaPeriod": "100ms",
+  "nodeStatusMaxImages": 50,
+  "maxOpenFiles": 1000000,
+  "contentType": "application/vnd.kubernetes.protobuf",
+  "kubeAPIQPS": 50,
+  "kubeAPIBurst": 100,
+  "serializeImagePulls": false,
+  "evictionHard": {
+    "imagefs.available": "15%",
+    "memory.available": "100Mi",
+    "nodefs.available": "10%",
+    "nodefs.inodesFree": "5%"
+  },
+  "evictionPressureTransitionPeriod": "5m0s",
+  "enableControllerAttachDetach": true,
+  "makeIPTablesUtilChains": true,
+  "iptablesMasqueradeBit": 14,
+  "iptablesDropBit": 15,
+  "featureGates": {
+    "APIPriorityAndFairness": true,
+    "CSIMigrationAWS": false,
+    "CSIMigrationAzureFile": false,
+    "CSIMigrationGCE": false,
+    "CSIMigrationvSphere": false,
+    "DownwardAPIHugePages": true,
+    "PodSecurity": true,
+    "RotateKubeletServerCertificate": true
+  },
+  "failSwapOn": true,
+  "memorySwap": {},
+  "containerLogMaxSize": "50Mi",
+  "containerLogMaxFiles": 5,
+  "configMapAndSecretChangeDetectionStrategy": "Watch",
+  "systemReserved": {
+    "cpu": "500m",
+    "memory": "1Gi"
+  },
+  "enforceNodeAllocatable": [
+    "pods"
+  ],
+  "volumePluginDir": "/etc/kubernetes/kubelet-plugins/volume/exec",
+  "logging": {
+    "format": "text",
+    "flushFrequency": 5000000000,
+    "verbosity": 2,
+    "options": {
+      "json": {
+        "infoBufferSize": "0"
+      }
+    }
+  },
+  "enableSystemLogHandler": true,
+  "shutdownGracePeriod": "0s",
+  "shutdownGracePeriodCriticalPods": "0s",
+  "enableProfilingHandler": true,
+  "enableDebugFlagsHandler": true,
+  "seccompDefault": false,
+  "registerNode": true,
+  "kind": "KubeletConfiguration",
+  "apiVersion": "kubelet.config.k8s.io/v1beta1"
+}
+```
+
+When we get `KubeletConfig` for each node, we will compare them, and
+only save the consistent result to a file at `/api/v1/nodes/proxy/configz`.
+
+In the CaC content, we just need to check one location `/api/v1/nodes/proxy/configz`
+for `KubeletConfig` default configuration.
+
+### User Stories
+
+* As an Site Reliability Engineer managing multiple clusters with many nodes, I'd like to run
+  compliance scans on my fleet in such a way that we have less compliance rule failures when
+  no reboot remediation has been applied.
+
+### API Extensions
+
+None is changed
+
+
+### Risks and Mitigations
+
+There are no risks or mitigations for this change.
+
+## Design Details
+
+### Open Questions [optional]
+
+
+1. Is `/api/v1/nodes/{NODE_NAME}/proxy/configz` a good place to check
+   default `KubeletConfig`?
+
+   Yes, it is a good place to check default `KubeletConfig`
+
+2. If we want to check the KubeletConfig through Kubernetes API, the rule
+   needs to change from node-rule to platform rule. Since platform rules
+   does not care about node type, it will require all nodes to have the 
+   same `KubeletConfig` configuration, wondering if that is a good idea?
```

Comment:
One more question: When you say we would require the same configuration do you mean that all configurations would really be 100% identical or that they would all be required to have the parameters we care about? Because if we care about the attributes only, then what if we wrote the rules in such a way that we check a list of kubeconfigs and require that all attributes at a certain yamlpaths are the same?

Reply:
they need to have the same parameters. I think that can work too.

Reply:
lets see if your idea with having a single pool and a single config under a special pool dir is doable. I like that more.

Reply:
Vincent's PR showed that his idea was doable and the code is nice.